src/Microsoft.VisualStudio.SolutionPersistence/Serializer/SlnV12/SlnFileV12Serializer.Reader.cs

```diff
@@ -545,6 +566,22 @@ private SolutionItemModel ReadProjectInfo(SolutionModel solution, ref StringToke
             }
             else
             {
+                // This should error, or remove any configuration associated with the duplicate id.
+                // However some old parsers would just ignore the duplicate id and continue.
+                if (duplicateItem is SolutionFolderModel duplicateFolder)
+                {
+                    duplicateFolder.Id = new Guid();
+                    this.tarnished = true;
+                }
+                else if (duplicateItem is SolutionProjectModel duplicateProject)
+                {
+                    projectId = Guid.NewGuid();
```

Comment:
I'm wondering if this should be a fallback if the project factory can't be asked to provide a Guid.

Reply:
Update to use the default slnx project id if possible.